tests/test_plugin_userinput.py

```diff
@@ -8,51 +8,67 @@
-        with pytest.raises(FatalPluginError) as cm:
-            testplugin.input_ask("test")
-        assert str(cm.value) == "This plugin requires user input, however it is not supported on this platform"
-
-        with pytest.raises(FatalPluginError) as cm:
-            testplugin.input_ask_password("test")
-        assert str(cm.value) == "This plugin requires user input, however it is not supported on this platform"
-
-    @pytest.mark.parametrize("console_input", [True], indirect=True)
-    def test_user_input_console(self, testplugin: _TestPlugin, console_input: ConsoleUserInputRequester):
-        assert testplugin.input_ask("username") == "username"
-        assert console_input.console.ask.call_args_list == [call("username: ")]
-
-        assert testplugin.input_ask_password("password") == "password"
-        assert console_input.console.askpass.call_args_list == [call("password: ")]
-
-    @pytest.mark.parametrize("console_input", [False], indirect=True)
-    def test_user_input_console_no_tty(self, testplugin: _TestPlugin, console_input: ConsoleUserInputRequester):
-        with pytest.raises(FatalPluginError) as cm:
-            testplugin.input_ask("username")
-        assert str(cm.value) == "User input error: no TTY available"
-
-        with pytest.raises(FatalPluginError) as cm:
-            testplugin.input_ask_password("username")
-        assert str(cm.value) == "User input error: no TTY available"
+@pytest.fixture()
+def console(request: pytest.FixtureRequest):
+    param = getattr(request, "param", {})
+
+    console = Mock()
+    if not param.get("failure", False):
+        console.ask.return_value = "username"
+        console.ask_password.return_value = "password"
+    else:
+        console.ask.side_effect = OSError("No input TTY available")
+        console.ask_password.side_effect = OSError("No input TTY available")
+
+    return console
+
+
+@pytest.fixture()
+def session(session: Streamlink, console: Mock):
+    user_input = ConsoleUserInputRequester(console)
+    session.set_option("user-input-requester", user_input)
+
+    return session
+
+
+@pytest.fixture()
+def testplugin(session: Streamlink):
+    return _TestPlugin(session, "http://example.com/stream")
+
+
+def test_session(session: Streamlink, console: Mock):
+    user_input = session.get_option("user-input-requester")
+    assert user_input
+    assert user_input.console is console
+
+
+def test_user_input_not_implemented(session: Streamlink, testplugin: _TestPlugin):
+    session.set_option("user-input-requester", None)
+
+    with pytest.raises(FatalPluginError) as cm:
+        testplugin.input_ask("test")
+    assert str(cm.value) == "This plugin requires user input, however it is not supported on this platform"
+
+    with pytest.raises(FatalPluginError) as cm:
+        testplugin.input_ask_password("test")
+    assert str(cm.value) == "This plugin requires user input, however it is not supported on this platform"
```

Comment:
Could we possibly use a better name for this? The name `test_user_input_not_implemented` plus the file name alluding to testing user input makes it seem like the plugin is at fault for somehow not accepting the input but the message then says the platform is at fault. Can we clarify it in some way to make it more consistent that the problem is system/platform input and not a problem with the plugin itself?

Reply:
I didn't touch any Plugin-specific APIs. These were left intact, and if they require changes, then in a different PR.

The error message you're commenting on comes from the `Plugin` class which uses the `UserInputRequester` interface and thus needs to do error handling, in which case it is raising its own `FatalPluginError` exception. This has nothing to do with any changes here.

I don't see any problem with the error message, but if you have a suggestion for an improvement, then please post it.

Reply:
Sounds good, let's leave it.